File: 0000_0100/82/82.cpp

// * Definition for singly-linked list.
struct ListNode {
    int val;
    ListNode* next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode* next) : val(x), next(next) {}
};


class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* dummy = new ListNode(0); // Create a dummy node
        dummy->next = head;
        ListNode* prev = dummy;
        ListNode* curr = dummy->next;

        while (curr) {
            if (curr->next && curr->val == curr->next->val) {
                int value = curr->val;
                while (curr && curr->val == value) {
                    curr = curr->next;
                }
                prev->next = curr;
            } else {
                prev = curr;
                curr = curr->next;
            }
        }

        return dummy->next;
    }
};

File: 0000_0100/82/82.py
# Definition for singly-linked list.

from typing import Optional


class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode()
        dummy.next = head
        prev = dummy
        current = dummy.next

        while current:
            if current.next and current.val == current.next.val:
                # dupe found skipppp allllll
                value = current.val
                while current and current.val == value:
                    current = current.next
                prev.next = current  # Link the last non dupe node to the next non dupe node

            else:
                prev = current
                current = current.next

        return dummy.next
